### PR TITLE
 Check the DO_JEDISNOWDA condition before adding snow DA analysis to the list

### LINKED ISSUE #2469: Archive Jobs Fail
### What is wrong?

gdasarch jobs fails on Hera due to:

ERROR: matchdir:  opendir error 2 opening directory gdas.20230403/00/analysis/snow/
*** Warning: No matching names located for 'gdas.20230403/00/analysis/snow/*[0000.sfc_data.tile1.nc](http://0000.sfc_data.tile1.nc/)'

 

### What should have happened?

It seems like the failure is due to the exglobal_archive.sh script attempting to copy files from ${COM_SNOW_ANALYSIS} directory to ${ARCDIR}, but the ${COM_SNOW_ANALYSIS} directory not exist. 

### What machines are impacted?

Hera

### Steps to reproduce

Run global workflow v17 archive jobs on Hera.
Global workflow commit:  59cdc0ee81926ee8dc7b8e544337bfc85130ad18

### Additional information

log file:  /scratch2/NCEPDEV/stmp3/Azadeh.Gholoubi/TestExperiment/MonthThompsonMP/logs/2023040218/gdasarch.log

### Do you have a proposed solution?

_No response_

### PR DESCRIPTION
**Pull request by jiaruidong2017**
# Description
This PR added checking the DO_JEDISNOWDA condition before adding snow DA analysis to the archive list. 

  Resolves #2469 
 
# Type of change

- Bug fix (fixes something broken)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# Checklist
- [ ] Any dependent changes have been merged and published
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

File: ush/hpssarch_gen.sh [hunk 1]
```diff
@@ -506,12 +506,14 @@ if [[ ${type} == "gdas" ]]; then
 
     [[ -s "${COM_CONF}/${head}letkfoi.yaml" ]] && echo "${COM_CONF/${ROTDIR}\//}/${head}letkfoi.yaml"
 
-    echo "${COM_SNOW_ANALYSIS/${ROTDIR}\//}/*0000.sfc_data.tile1.nc"
-    echo "${COM_SNOW_ANALYSIS/${ROTDIR}\//}/*0000.sfc_data.tile2.nc"
-    echo "${COM_SNOW_ANALYSIS/${ROTDIR}\//}/*0000.sfc_data.tile3.nc"
-    echo "${COM_SNOW_ANALYSIS/${ROTDIR}\//}/*0000.sfc_data.tile4.nc"
-    echo "${COM_SNOW_ANALYSIS/${ROTDIR}\//}/*0000.sfc_data.tile5.nc"
-    echo "${COM_SNOW_ANALYSIS/${ROTDIR}\//}/*0000.sfc_data.tile6.nc"
+    if [[ ${DO_JEDISNOWDA} = "YES" ]]; then
```
Comment (aerorahul) [inline ush/hpssarch_gen.sh]:
```suggestion
    if [[ "${DO_JEDISNOWDA:-}" == "YES" ]]; then
```

**Review by aerorahul — CHANGES_REQUESTED** [review]
Thank you for adding the conditional.  This will work.

However, in `config.base`, the variable is named `DO_JEDILANDDA`.  It should be renamed to `DO_JEDISNOWDA`.  Would you mind changing that variable as well to be `DO_JEDISNOWDA` instead of `DO_JEDILANDDA`?

File: ush/hpssarch_gen.sh (hunk 1 above)
Comment (jiaruidong2017) [reply ush/hpssarch_gen.sh]: Thanks @aerorahul and done the changes as suggested.

Comment (jiaruidong2017) [reply ush/hpssarch_gen.sh]: Also changed the `DO_JEDILANDDA` to `DO_JEDISNOWDA`.

**Review by aerorahul — APPROVED** [review]
looks good.
At somepoint we should review what needs to be archived for the `sfc_data` files.  We have `sfcanl_data` tile files also being archived.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Pull request merged**